jasmine/spec/inverted-index-test.js

```diff
@@ -0,0 +1,170 @@
+const invalidStructure = require('./dummy-data/file5.json');
+const invalidContent = require('./dummy-data/file4.json');
+const nonString = require('./dummy-data/file3.json');
+var Index = require('../../src/inverted-index.js');
+
+function getFile(filename, cb) {
+  const theResult = [filename];
+  cb(theResult);
+}
+describe('Read Book data', function() {
+  const index = new Index();
+  describe('When I upload a JSON file', () => {
+    it('It should checks if its a valid JSON array', (done) => {
+      expect(index.saveUploads('invalidStructure.json', invalidStructure)).toBeFalsy();
+      done();
+    });
+    it('It should check if the file is empty', (done) => {
+      expect(index.saveUploads('empty.json', empty)).toBeFalsy();
+      done();
+    });
+    it('It should check if Its property values are strings', (done) => {
+      expect(index.saveUploads('nonString.json', nonString)).toBeFalsy();
+      done();
+    });
+    it('get json database should return the saved content', (done) => {
+      index.saveUploads('valid1.json', valid1);
+      expect(Object.keys(index.getJsonDatabase()).length).toEqual(1);
+      done();
+    });
+    it('getfilename should return the filenames of the saved contents', (done) => {
+      index.saveUploads('valid1.json', valid1);
+      expect(index.getFilenames()).toEqual(['valid1.json']);
+      done();
+    });
+  });
+});
+describe('Populate Index', () => {
+  const index = new Index();
+  index.saveUploads('valid1.json', valid1);
+  index.saveUploads('valid2.json', valid1);
+  index.createIndex('valid1.json');
+
+  describe('When I Upload a File', () => {
+    it('the index is created once the JSON file has been read', (done) => {
+      expect(typeof index.getIndex('valid1.json')).toEqual('object');
+      done();
+    });
+
+    it('the created index should be an accurate one', (done) => {
+      expect(index.getIndex('valid1.json').alice[0]).toEqual(0);
+      expect(index.getIndex('valid1.json').lord[0]).toEqual(1);
+      expect(index.getIndex('valid1.json').a[1]).toEqual(1);
+      done();
+    });
+
+    it('the created index should be an inverted index', (done) => {
+      let verdict = true;
+      const indexContent = index.indexFile['valid1.json'];
+
+      for (value in indexContent) {
+        if (!Array.isArray(indexContent[value]) || isNaN(indexContent[value][0])) {
+          verdict = false;
+        }
+      }
+      expect(verdict).toEqual(true);
+      done();
+    });
+    it('it should not overwrite the index', (done) => {
+      const indexBefore = index.getIndex('valid1.json');
+      index.createIndex('valid2.json');
+      const indexAfter = index.getIndex('valid1.json');
+      expect(indexBefore).toEqual(indexAfter);
+      done();
+    });
+  });
+});
+
+describe('Search Index', () => {
+  const index = new Index();
+  index.saveUploads('valid1.json', valid1);
+  index.createIndex('valid1.json');
+
+  describe('should return the correct result when searched', () => {
```

Comment:
Doesn't seem like you followed the testing conventions here. Your `describe` block is saying what your `it` block is supposed to say. You may want to look at this again and restructure it